ocaml-lsp-server/src/custom_requests/req_jump_typed_hole.ml

```diff
@@ -0,0 +1,136 @@
+open Import
+
+let capability = "handleJumpTypedHole", `Bool true
+let meth = "ocamllsp/jumpTypedHole"
+
+module Request_params = struct
+  type t =
+    { text_document : TextDocumentIdentifier.t
+    ; position : Position.t
+    ; range : Range.t option
+    ; direction : [ `Next | `Prev ]
+    }
+
+  let create ?(direction = `Next) ?range ~text_document ~position () =
+    { text_document; position; direction; range }
+  ;;
+
+  let yojson_of_direction = function
+    | `Next -> `String "next"
+    | `Prev -> `String "prev"
+  ;;
+
+  let yojson_of_t { text_document; position; direction; range } =
+    match TextDocumentIdentifier.yojson_of_t text_document with
+    | `Assoc assoc ->
+      let position = "position", Position.yojson_of_t position in
+      let range =
+        ( "range"
+        , match range with
+          | None -> `Null
+          | Some r -> Range.yojson_of_t r )
+      in
+      let direction = "direction", yojson_of_direction direction in
+      `Assoc (direction :: position :: range :: assoc)
+    | _ -> (* unreachable *) assert false
+  ;;
+
+  let direction_of_yojson json =
+    let open Yojson.Safe.Util in
+    let dir = json |> member "direction" |> to_string in
+    match String.lowercase_ascii dir with
+    | "prev" -> `Prev
+    | _ -> `Next
+  ;;
+
+  let t_of_yojson json =
+    let open Yojson.Safe.Util in
+    let text_document = TextDocumentIdentifier.t_of_yojson json in
+    let position = json |> member "position" |> Position.t_of_yojson in
+    let direction = direction_of_yojson json in
+    let range = json |> member "range" |> to_option Range.t_of_yojson in
+    { text_document; position; direction; range }
+  ;;
+end
+
+type t = Range.t option
+
+let t_of_yojson opt =
+  let open Yojson.Safe.Util in
+  to_option Range.t_of_yojson opt
+;;
+
+let yojson_of_t = function
+  | None -> `Null
+  | Some range -> Range.yojson_of_t range
+;;
+
+let holes_in_range range holes =
+  match range with
+  | None -> holes
+  | Some range -> List.filter ~f:(Range.contains range) holes
+;;
+
+let find_prev_hole ~range ~position holes =
+  let holes = holes_in_range range holes in
+  Base.List.fold_until
```

Comment:
I expect that Base is already open right ? (see `find_next_hole`)

```suggestion
  List.fold_until
```

Reply:
No, `Base` is not part of `Import` so it is not opened.